File: sample/app/src/main/java/com/igaworks/dfinerysample/BaseFirebaseMessagingService.java
package com.igaworks.dfinerysample;

import android.util.Log;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.igaworks.dfinery.Dfinery;
import com.igaworks.dfinery.DfineryProperties;

public class BaseFirebaseMessagingService extends FirebaseMessagingService {
    private static final String TAG = "BaseFirebaseMessagingService";
    @Override
    public void onNewToken(@NonNull String registrationId) {
        super.onNewToken(registrationId);
        DfineryProperties.setPushToken(registrationId);
    }

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        if(Dfinery.getInstance().handleRemoteMessage(getApplicationContext(), remoteMessage.getData())){
            Log.d(TAG, remoteMessage.getData().toString());
        }else{
            //This is not a push notification sent from Dfinery.
        }
    }
}

File: sample/kotlin/app/src/main/java/com/igaworks/dfinerykotlinsample/BaseFirebaseMessagingService.kt
package com.igaworks.dfinerykotlinsample

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.igaworks.dfinery.Dfinery
import com.igaworks.dfinery.DfineryProperties

class BaseFirebaseMessagingService: FirebaseMessagingService() {
    val TAG: String? = BaseFirebaseMessagingService::class.java.canonicalName

    override fun onNewToken(registrationId: String) {
        super.onNewToken(registrationId)
        DfineryProperties.setPushToken(registrationId)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        if (Dfinery.getInstance().handleRemoteMessage(applicationContext, remoteMessage.data)) {
            Log.d(TAG, remoteMessage.data.toString())
        } else {
            //This is not a push notification sent from Dfinery.
        }
    }
}